modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ValidationOnNodeJoinUtils.java

```diff
@@ -171,12 +179,26 @@ public class ValidationOnNodeJoinUtils {
                     }
                 }
 
-                DynamicCacheDescriptor locDesc = cacheDescProvider.apply(cacheInfo.cacheData().config().getName());
+                //check is performed only on affinity nodes for the cache requested to start
+                if (CU.affinityNode(ctx.discovery().localNode(), joiningNodeCacheCfg.getNodeFilter())) {
```

Comment:
We will skip the check if coordinator node is filtered by affinity.

Reply:
Good catch, I removed this check and marked to ignore a jUnit test relying in this behavior.